### PR TITLE
Re-enable cuspatial tests and dependency.

### PR DESCRIPTION
**Pull request by thomcom**
## Description
A poorly understood cuproj->cuspatial dependency required that cuspatial be removed from cuproj's tests during the 23.08->23.10 merge rebase. This PR re-enables those tests.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by isVoid** [comment]
I was a bit out of loop of this in 23.08, in which PR did you disabled the cuproj tests in 23.08?

**Comment by thomcom** [comment]
They weren't disabled in 23.08, they were disabled in the 23.08 -> 23.10 merge commit because of the dependency that cuproj has on cuspatial, which caused ci to load the cuspatial-nightly build instead of the in-PR build of cuspatial, running cuproj 23.10 tests against cuspatial 23.08.

File: dependencies.yaml [hunk 1]
```diff
@@ -442,3 +442,4 @@ dependencies:
           - pytest-cov
           - pytest-xdist
           - geopandas>=0.11.0
+          - cuspatial=23.10.*
```
Comment (harrism) [inline dependencies.yaml:445]:
```suggestion
          - cuspatial==23.10.*
```

And re-run pre-commit.

**Comment by harrism** [comment]
@thomcom your .devcontainer folder doesn't match the latest for 23.10, which has separate folders: 

```
.devcontainer/
├── cuda11.8-conda
│   └── devcontainer.json
├── cuda11.8-pip
│   └── devcontainer.json
├── cuda12.0-conda
│   └── devcontainer.json
├── cuda12.0-pip
│   └── devcontainer.json
├── Dockerfile
└── README.md
```

**Comment by harrism** [comment]
Please remember:

/ok to test

File: dependencies.yaml [hunk 2]
```diff
@@ -347,6 +347,7 @@ dependencies:
           - pytest-cov
           - pytest-xdist
           - geopandas>=0.11.0
+          - cuspatial==23.10.*
       - output_types: [requirements, pyproject]
         packages:
           - pyproj
```
Comment (harrism) [inline dependencies.yaml:353]:
```suggestion
          - pyproj>=3.6.0,<3.7a0
```

Apparently not specifying a loose pin causes conda to implicitly use a tight pin to the latest version. Doing this should avoid the problem of pinning to an older libtiff that proj 9.2 uses, and caused solver problems for us. Am I right @raydouglass ? See https://github.com/rapidsai/cuspatial/pull/1276

**Review by harrism — CHANGES_REQUESTED** [review]
One dependency request. Remember to run `pre-commit run` before committing when you make this change.

**Review by harrism — APPROVED** [review]

File: conda/environments/all_cuda-120_arch-x86_64.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ dependencies:
 - cudf==23.10.*
 - cuml==23.10.*
 - cupy>=12.0.0
+- cuspatial==23.10.*
```
Comment (bdice) [inline conda/environments/all_cuda-120_arch-x86_64.yaml:20]: This is coming from `dependencies.yaml`. We don't want cuspatial as a requirement of its own development environment.

**Review by bdice — APPROVED** [review]

**Review by ajschmidt8 — APPROVED** [review]

**Review by raydouglass — APPROVED** [review]

**Pull request merged**